gix-ref/src/store/file/log/line.rs

```diff
@@ -123,34 +123,35 @@ pub mod decode {
         }
     }
 
-    fn message<'a, E: ParseError<&'a [u8]>>(i: &'a [u8]) -> IResult<&'a [u8], &'a BStr, E> {
+    fn message<'a, E: ParserError<&'a [u8]>>(i: &'a [u8]) -> IResult<&'a [u8], &'a BStr, E> {
         if i.is_empty() {
             Ok((&[], i.as_bstr()))
         } else {
-            terminated(take_while(|c| c != b'\n'), opt(tag(b"\n")))(i).map(|(i, o)| (i, o.as_bstr()))
+            terminated(take_while(0.., |c| c != b'\n'), opt(b"\n"))
+                .parse_next(i)
+                .map(|(i, o)| (i, o.as_bstr()))
         }
     }
 
-    fn one<'a, E: ParseError<&'a [u8]> + ContextError<&'a [u8]>>(bytes: &'a [u8]) -> IResult<&[u8], LineRef<'a>, E> {
-        let (i, (old, new, signature, message_sep, message)) = context(
-            "<old-hexsha> <new-hexsha> <name> <<email>> <timestamp> <tz>\\t<message>",
-            tuple((
-                context("<old-hexsha>", terminated(hex_hash, tag(b" "))),
-                context("<new-hexsha>", terminated(hex_hash, tag(b" "))),
-                context("<name> <<email>> <timestamp>", gix_actor::signature::decode),
-                opt(tag(b"\t")),
-                context("<optional message>", message),
-            )),
-        )(bytes)?;
+    fn one<'a, E: ParserError<&'a [u8]> + AddContext<&'a [u8]>>(bytes: &'a [u8]) -> IResult<&[u8], LineRef<'a>, E> {
+        let (i, (old, new, signature, message_sep, message)) = (
+            terminated(hex_hash, b" ").context("<old-hexsha>"),
+            terminated(hex_hash, b" ").context("<new-hexsha>"),
+            gix_actor::signature::decode.context("<name> <<email>> <timestamp>"),
+            opt(b"\t"),
+            message.context("<optional message>"),
+        )
+            .context("<old-hexsha> <new-hexsha> <name> <<email>> <timestamp> <tz>\\t<message>")
+            .parse_next(bytes)?;
 
         if message_sep.is_none() {
             if let Some(first) = message.first() {
                 if !first.is_ascii_whitespace() {
```

Comment:
This is a little confusing.  We check for the separator being a tab but we actually allow any whitespace separator.  

I feel like it should either be:
```rust
opt((
    b"\t",
    message
))
```
or
```rust
opt((
    one_of(|b| b.is_ascii_whitespace()),
    message
))
```

unless we are intentionally capturing non-tab whitespace separators to be a part of `message`.

Reply:
Sorry for the late reply. There should always be a test for special cases, and if nothing breaks with the simplified version this is probably an oversight.